src/emulator/dataconnectEmulator.ts

```diff
@@ -157,8 +220,27 @@ export class DataConnectEmulator implements EmulatorInstance {
       this.logger.log("DEBUG", "No Postgres connection string found, not connecting to Postgres");
       return false;
     }
-    await this.emulatorClient.configureEmulator({ connectionString, database, serviceId });
-    return true;
+    // The Data Connect emualtor does not immediately start listening after started
+    // so we retry this call with a brief backoff
```

Comment:
The emulator prints off a "listening" log for each listen spec at INFO level btw, which should be available after https://github.com/firebase/firebase-tools/pull/7333 . It may provide a better signal than retrying. This can be done in a follow up PR of course.

Reply:
Added a todo for this, good call.